lib/core/bucketEntries/BucketEntry.ts

```diff
@@ -1,12 +1,18 @@
-import { Bucket } from "../buckets/Bucket";
-import { Frame } from "../frames/Frame";
+import { Bucket } from '../buckets/Bucket';
+import { Frame } from '../frames/Frame';
 
 export interface BucketEntry {
   id: string;
+  // Check if frame can be id or frame itself:
```

Comment:
This will happen when a populate is used, therefore it is the responsibility of the Repository to type that situation. Here we are just modeling the data structures as they are raw